operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/internal/CustomResourceEventSource.java

```diff
@@ -98,49 +98,38 @@ public void eventReceived(Watcher.Action action, T customResource) {
       return;
     }
 
-    if (!skipBecauseOfGeneration(customResource)) {
-      eventHandler.handleEvent(new CustomResourceEvent(action, customResource, this));
-      markLastGenerationProcessed(customResource);
+    boolean eventAccepted;
+    if (oldResource == null || customResource.isMarkedForDeletion()) {
+      eventAccepted = true;
     } else {
-      log.debug(
-          "Skipping event handling resource {} with version: {}",
-          getUID(customResource),
-          getVersion(customResource));
-    }
-  }
+      final var configuration = controller.getConfiguration();
 
-  private void markLastGenerationProcessed(T resource) {
-    if (controller.getConfiguration().isGenerationAware()
-        && resource.hasFinalizer(controller.getConfiguration().getFinalizer())) {
-      lastGenerationProcessedSuccessfully.put(
-          KubernetesResourceUtils.getUID(resource), resource.getMetadata().getGeneration());
-    }
-  }
+      // if the finalizer has not been set, the operator logic will add it so there will be
+      // events generated by the operator itself that won't lead to a new generation so such
+      // events should not be filtered
+      final CustomResourceEventFilter<T> finalizerPredicate =
```

Comment:
Why are we checking this with predicate? It would be much simpler and easy to read with just checking directly the field with in a method.

Reply:
So that the same logic is applied everywhere it's needed instead of having variations of the same code.